sway-parse/src/attribute.rs

```diff
@@ -1,13 +1,78 @@
+use crate::priv_prelude::{Peek, Peeker};
 use crate::{Parse, ParseBracket, ParseErrorKind, ParseResult, ParseToEnd, Parser, ParserConsumed};
 
 use sway_ast::attribute::{Annotated, Attribute, AttributeDecl};
-use sway_ast::brackets::Parens;
-use sway_ast::keywords::{HashToken, StorageToken};
-use sway_types::Ident;
+use sway_ast::brackets::{Parens, SquareBrackets};
+use sway_ast::keywords::{CommaToken, HashToken, StorageToken, Token};
+use sway_ast::punctuated::Punctuated;
+use sway_ast::token::{DocComment, DocStyle};
+use sway_types::{Ident, Span};
+
+impl Peek for DocComment {
+    fn peek(peeker: Peeker<'_>) -> Option<DocComment> {
+        peeker
+            .peek_doc_comment()
+            .ok()
+            .map(|doc_comment| doc_comment.clone())
+        // .map(DocComment::clone)
+    }
+}
+
+impl Parse for DocComment {
+    fn parse(parser: &mut Parser) -> ParseResult<DocComment> {
+        match parser.take::<DocComment>() {
+            Some(doc_comment) => Ok(doc_comment),
+            None => Err(parser.emit_error(ParseErrorKind::ExpectedDocComment)),
+        }
+    }
+}
 
 impl<T: Parse> Parse for Annotated<T> {
     fn parse(parser: &mut Parser) -> ParseResult<Self> {
         let mut attribute_list = Vec::new();
+
+        // Parse outer doc comment lines
+        let mut doc_lines = Vec::new();
+        loop {
+            if matches!(
+                parser.peek::<DocComment>(),
+                Some(DocComment {
+                    doc_style: DocStyle::Outer,
+                    ..
+                })
+            ) {
+                let doc_comment = parser.parse::<DocComment>()?;
+                let span = doc_comment.span.clone().as_str().to_string();
+                let line = span.trim_start_matches("///").to_string();
+                let line = line.trim().to_string();
+                doc_lines.push(line);
+            } else {
+                break;
+            }
+        }
+
+        // Join doc comments and add as an AttributeDecl
```

Comment:
Should we even merge these? Maybe every doc comment should be a separate attribute?

Reply:
Seems like Rust doesn't so not merging.